src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java

```diff
@@ -0,0 +1,154 @@
+package com.minecolonies.coremod.colony.requestsystem.resolvers.core;
+
+import com.google.common.base.Functions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.location.ILocation;
+import com.minecolonies.api.colony.requestsystem.manager.IRequestManager;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.request.RequestState;
+import com.minecolonies.api.colony.requestsystem.requestable.Stack;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.crafting.IRecipeStorage;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.blockout.Log;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public abstract class AbstractCraftingRequestResolver extends AbstractBuildingDependentRequestResolver<Stack>
+{
+    public AbstractCraftingRequestResolver(
+      @NotNull final ILocation location,
+      @NotNull final IToken<?> token)
+    {
+        super(location, token);
+    }
+
+    @Override
+    public TypeToken<? extends Stack> getRequestType()
+    {
+        return TypeToken.of(Stack.class);
+    }
+
+    @Override
+    public boolean canResolveForBuilding(
+      @NotNull final IRequestManager manager, @NotNull final IRequest<? extends Stack> request, @NotNull final AbstractBuilding building)
+    {
+        if (createsCraftingCycle(manager, request, request))
+            return false;
+
+        return building instanceof AbstractBuildingWorker && canBuildingCraftStack((AbstractBuildingWorker) building, request.getRequest().getStack());
+    }
+
+    protected boolean createsCraftingCycle(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends Stack> request, @NotNull final IRequest<? extends Stack> target)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 181 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)